community_supplied/Linecard/netconf-fpc-cpu-stats.rule

```diff
@@ -0,0 +1,261 @@
+            */
+            keys slot;
+            sensor fpc-sensor {
+			    synopsis "FPC CPU Memory sensor initialization";
+                description "FPC iAgent sensor to collect data from network device";
+                iAgent {
+                    file fpc-cpu-memory-stats.yml;
+                    table FPCCpuMemoryStatsTable;
+                    frequency 300s;              
+                }
+            }
+            /*
+            * Fields defined with variables, dynamic thresholds and constant values
+            */
+            field dt-buffer-util {
+                formula {
+                    dynamic-threshold {
+                        algorithm k-means;
+                        learning-period 7d;
+                        pattern-periodicity 1d;
+                        field-name "$memory-buffer-utilization";
+                    }
+                }
+                type integer;
+                description "dynamic threshold value for memory buffer utilization";
+            }
+            field dt-cpu-total {
+                formula {
+                    dynamic-threshold {
+                        algorithm k-means;
+                        learning-period 7d;
+                        pattern-periodicity 1d;
+                        field-name "$cpu-total";
+                    }
+                }
+                type integer;
+                description "dynamic threshold value for CPU utilization";
+            }
+            field dt-heap-util {
+                formula {
+                    dynamic-threshold {
+                        algorithm k-means;      
+                        learning-period 7d;
+                        pattern-periodicity 1d;
+                        field-name "$memory-heap-utilization";
+                    }
+                }
+                type integer;
+                description "dynamic threshold value for memory heap utilization";
+            }
+            field slot {
+                sensor fpc-sensor {
+                    where "slot =~ /{{slot-name}}/";
+                    path slot;
+                }
+                type string;
+                description "fpc slot number";
+            }
+            field threshold {
+                constant {
+                    value "{{static-threshold}}";
+                }
+                type integer;
+                description "Threshold value for memory & CPU utilization values";
+            }
+			/*
+            * Anomaly detection logic for buffer memory utilization.
+            */
+            trigger buffer-memory-utilization {
+                frequency 300s;
+				/*
+                * Sets color to red and sends out an anomaly notification when
+                * memory buffer utilization count increases more than threshold static ($threshold) value
+                * 
+                */
+                term is-buffer-memory-utilization-is-abnormal {

```

Comment:
The term names shouldn't have to "is" in them. 
It should be is-buffer-memory-utilization-abnormal..

Reply:
Fixed